pi-hole/docker-pi-hole
# Introduce additional docker tags for the debian version.

Introduce additional docker tags for the Debian version, allows building both stretch and buster images.

## Description

* Added new docker tag variations to specify the Debian version (`stretch`, and `buster`).
* **Arch images are always as specific as possible: `pihole/pihole:master-amd64-stretch`. This is a bit of a breaking change, as there are no more `pihole/pihole:master-amd64` tags... not sure if that is a problem?**
* Multiarch images have both the specific Debian version tags as well as the generic non-debian tags: `pihole/pihole:master-stretch` & `pihole/pihole:master`
* Currently, the non-specific tags point to the 'stretch' images. Eventually it can be migrated to `buster`.
* Use GitHub actions to do the builds. Although the script names include 'gh-actions' to differentiate them from the 'circle' scripts, there is zero logic that is specific to Github (ie. no Github environment variables).
* `armhf:buster` & `arm64:buster` has an issue with `ip route get`. I think the issue is related to `qemu`, but I'm not sure. Update the `validate_env` function to only use `ip route get` if `nc` reports something strange.
* Delete circle-ci job since it will fail with these changes. I've not included that here. I can delete the files if wanted.
* This introduces `buster` and `stretch` tags, other then getting them to build and pass tests, I haven't done any further verification of the `buster` images


## Motivation and Context

* Opens a migration path for eventually having buster images being the primary Debian version and future debian versions
* Uses Github Actions which are a little friendlier and easier to find. Doesn't require a circle-ci account.

## How Has This Been Tested?

* Locally and Github Actions.
* See the images build by my Github actions: https://hub.docker.com/repository/docker/lightswitch05/pihole
* See my Github actions: https://github.com/lightswitch05/docker-pi-hole/runs/746208461?check_suite_focus=true

## Types of changes


## Checklist:
